bindings/java/src/main/com/apple/foundationdb/EventKeeper.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * TransactionTimer.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apple.foundationdb;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A device for externally instrumenting the FDB java driver, for monitoring
+ * purposes.
+ *
+ * Note that implementations as expected to be thread-safe, and may be manipulated
+ * from multiple threads even in nicely single-threaded-looking applications.
+ */
+public interface EventKeeper {
+
+    /**
+     * Count the number of events which occurred.
+     * 
+     * @param event the event which occurred
+     * @param amt the number of times that even occurred
+     */
+    void count(Event event, long amt);
+
+    /**
+     * Convenience method to add 1 to the number of events which occurred.
+     * 
+     * @param event the event which occurred.
+     */
+    default void increment(Event event) {
+        count(event, 1L);
+    }
+
+    /**
+     * Count the time taken to perform an event, in nanoseconds.
+     * 
+     * Note that {@code event.isTimeEvent()} should return true here.
+     * 
+     * @param event the event which was timed (the event should be a time event).
+     * @param nanos the amount of time taken (in nanoseconds)
+     */
+    void timeNanos(Event event, long nanos);
+
+    /**
+     * Count the time taken to perform an action, in the specified units.
+     * 
+     * Note that {@code event.isTimeEvent()} should return true.
+     * 
+     * @param event the event which was timed.
+     * @param duration the time taken
+     * @param theUnit the unit of time in which the time measurement was taken
+     */
+    default void time(Event event, long duration, TimeUnit theUnit){
```

Comment:
why not use https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html  everywhere? cause it by itself is less memory efficient the primitive?

Reply:
you could, I suppose. My approach to this is partly habit (`Duration` is a Java 8 class, so all my Java 7 code had to use this pattern) and partly because the pattern 
```
start = System.nanoTime();
//do stuff
keeper.time(System.nanoTime()-start,TimeUnit.NANOSECONDS);
```
has a pretty natural, clean flow.

in the hottest of hot loops, the extra object creation of making `Duration` fields could add up, but I didn't make any benchmarks to guide that particular decision.